virttest/virsh.py

```diff
@@ -621,6 +624,93 @@ def kosher_args(remote_ip, uri):
         return True not in all_false
 
 
+class EventNotFoundError(Exception):
+    """
+    Error when certain event cannot be found.
+    """
+
+    def __init__(self, details=''):
+        self.details = details
+        Exception.__init__(self)
+
+    def __str__(self):
+        return str(self.details)
+
+
+class EventTracker(object):
+
+    @staticmethod
+    def start_get_event(vm_name):
+        """
+        Use a virsh session with subcommand 'event' to catch events
+
+        :param vm_name: name of the vm to be catched
+        :return: the virsh session with 'event'
+        """
+        virsh_session = aexpect.ShellSession(VIRSH_EXEC)
+        event_cmd = 'event %s --all --loop' % vm_name
```

Comment:
In many cases, we need specify event_type ,and timeout as constraints, can this implement handle them gracefully?

Reply:
Could you provide examples? How gracefully did you mean?

Reply:
> Could you provide examples? How gracefully did you mean?

At least, it need allow user to pass in event_type and timeout parameters  as command line, rather than hardcode --loop when start  virsh event

Reply:
But why, why is --loop so bad? I think it's pretty good to logging out other events as output when there might be, it's easier to debug, don't you think?

Reply:
I think we need support both --loop --event-type and  --event-type  --timeout ,this is test case requirement.

Reply:
@chunfuwen though I don't agree, but how do you propose we to do that to support?

Reply:
how about adding three additional parameters: one is event_type ,second is option(--loop or --timeout), and third one is **kargs(e.g provide timeout value) ?  I think there are lots of examples in virsh.py how to compose them together.

Reply:
I don't think it's quite necessary right now. But feel free to perfect the implementation in the future. Thanks.

Reply:
I would like this utility can implement --event-type --timeout rather than --loop if it need make choice . 
If we look at detach_device, attach_device, or  even change-media  currently n virsh.py, what they need is just to wait for certain type event with specific timeout value.

Reply:
@chunfuwen sure, feel free to do the modification after it got merged. Thanks.